src/main/fulcro/incubator/pessimistic_mutations.cljc

```diff
@@ -32,44 +46,62 @@
   [{:keys [ast ref]}]
   (when (:query ast)
     (log/error "You should not use mutation joins (returning) with `pmutate!`. Use the params of `pmutate!` instead."))
-  (some-> ast
-    (update :params dissoc ::returning ::target)
-    (cond-> (:query ast) (update :query vary-meta dissoc :component))
-    (mutations/with-target (conj ref ::mutation-response-swap))
-    (vary-meta assoc ::pessimistic-mutation true)))
+  (let [mutation-response-key (get-params-mutation-response-key ast)
+        mutation-response-swap-key (mutation-response-swap-keyword mutation-response-key)]
+    (some-> ast
+      (update :params dissoc ::returning ::target)
+      (cond-> (:query ast) (update :query vary-meta dissoc :component))
+      (mutations/with-target (conj ref mutation-response-swap-key))
+      (vary-meta assoc ::pessimistic-mutation true))))
 
 (defn- pessimistic-mutation?
   [ast]
   (-> ast meta ::pessimistic-mutation))
 
+(defn mutation-response-with-key
+  "Retrieves the mutation response from `this-or-props` (a component or props).  Can also be used against the state-map with an ident."
```

Comment:
maybe we can add information about the specifics on sending the mutation response key in these docs